aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -73,177 +73,278 @@ void TensorIterator::reorder_dimensions() {
   permute_dimensions(perm_);
 }
 
-// Returns the first non-CPU device, if present, or the CPU
-// if all operands are on the CPU.
-Device compute_device(at::ArrayRef<OperandInfo> operands) {
-  for (auto& op : operands) {
-    if (op.tensor.defined() && !op.tensor.device().is_cpu()) {
-      return op.tensor.device();
-    }
-  }
-
-  return kCPU;
-}
-
-static std::tuple<Device, ScalarType, bool> compute_common_type_(at::ArrayRef<OperandInfo> operands) {
-  // See [Result type computation] in TensorIterator.h
-  auto device = compute_device(operands);
-  auto common_type = ScalarType::Undefined;
-  bool all_same_type = true;
-  for (const auto& op: operands){
-    if (!op.tensor.defined()) continue;
-    //don't handle scalars
-    if (op.tensor.dim() > 0){
-      ScalarType current = op.current_dtype;
-      if (current == ScalarType::Undefined){
-        all_same_type = false;
-        break;
-      }
-      if (common_type == ScalarType::Undefined) common_type = current;
-      if (common_type != current) {
-        all_same_type = false;
-        break;
-      }
-    } else {
-      all_same_type = false;
-      break;
-    }
-  }
-  if (all_same_type) {
-    return std::make_tuple(device, common_type, true);
+// See [Result type computation] in TensorIterator.h
+// NOTE: This is (or at least, was) slow, and compute_types() (below) attempts
```

Comment:
"was" --> so what was this previously? `compute_common_type_`?

Reply:
I should improve the clarity of this comment. @ngimel suggested avoiding this path because it's been slow in the past, but the speed may have improved.